src/Microsoft.AspNet.Http.Abstractions/QueryString.cs

```diff
@@ -231,7 +231,7 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return (_value != null ? _value.GetHashCode() : 0);
+            return _value?.GetHashCode() ?? 0;
```

Comment:
It does look like it's doing what you listed. Nevermind, I guess I was just confused about how null coalescing works for nullable types.